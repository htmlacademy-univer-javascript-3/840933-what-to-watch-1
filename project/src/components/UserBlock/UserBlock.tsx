import { Link, useNavigate } from 'react-router-dom';

import { useAppDispatch, useAppSelector } from '../../hooks';
import { logoutAction } from '../../api/apiActionUser';
import { AuthorizationStatus } from '../../enums/auth.enum';
import { AppRoute } from '../../enums/route.enum';
import {
  getAuthorizationStatus,
  getUser,
} from '../../store/userReducer/selector';

export const UserBlock = () => {
  const dispatch = useAppDispatch();
  const navigate = useNavigate();
  const authorizationStatus = useAppSelector(getAuthorizationStatus);
  const user = useAppSelector(getUser);

  return (
    <ul className="user-block">
      {authorizationStatus === AuthorizationStatus.Auth ? (
        <>
          <li className="user-block__item">
            <div
              className="user-block__avatar"
              onClick={() => navigate('/mylist')}
            >
              <img
                src={user?.avatarUrl}
                alt="User avatar"
                width="63"
                height="63"
              />
            </div>
          </li>
          <li className="user-block__item">
            <Link
              onClick={() => dispatch(logoutAction())}
              className="user-block__link"
              to={'/'}
            >
              Sign out
            </Link>
          </li>
        </>
      ) : (
        <li className="user-block__item">
          <Link to={AppRoute.LOGIN_ROUTE} className="user-block__link">
            Sign in
          </Link>
        </li>
      )}
    </ul>
  );
};
