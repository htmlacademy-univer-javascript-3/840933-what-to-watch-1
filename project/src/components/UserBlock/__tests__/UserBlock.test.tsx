import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { configureMockStore } from '@jedmao/redux-mock-store';
import '@testing-library/jest-dom/extend-expect';

import { UserBlock } from '../UserBlock';
import { AuthorizationStatus } from '../../../enums/auth.enum';

const initialEntries = ['/'];
const mockStore = configureMockStore();

describe('<UserBlock />', () => {
  it('should render sign out when user has auth status', () => {
    const store = mockStore({
      user: { authorizationStatus: AuthorizationStatus.Auth },
    });

    const view = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={initialEntries}>
          <UserBlock />
        </MemoryRouter>
      </Provider>
    );

    expect(view).toMatchSnapshot();
    expect(screen.getByText('Sign out')).toBeInTheDocument();
  });

  it('should render sign in when user has no auth status', () => {
    const store = mockStore({
      user: { authorizationStatus: AuthorizationStatus.NoAuth },
    });

    const view = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={initialEntries}>
          <UserBlock />
        </MemoryRouter>
      </Provider>
    );

    expect(view).toMatchSnapshot();
    expect(screen.getByText('Sign in')).toBeInTheDocument();
  });
});
