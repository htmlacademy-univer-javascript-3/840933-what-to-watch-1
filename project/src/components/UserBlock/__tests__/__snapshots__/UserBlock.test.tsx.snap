// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserBlock /> should render sign in when user has no auth status 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="user-block"
      >
        <li
          class="user-block__item"
        >
          <a
            class="user-block__link"
            href="/login"
          >
            Sign in
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="user-block"
    >
      <li
        class="user-block__item"
      >
        <a
          class="user-block__link"
          href="/login"
        >
          Sign in
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<UserBlock /> should render sign out when user has auth status 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="user-block"
      >
        <li
          class="user-block__item"
        >
          <div
            class="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              width="63"
            />
          </div>
        </li>
        <li
          class="user-block__item"
        >
          <a
            class="user-block__link"
            href="/"
          >
            Sign out
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="user-block"
    >
      <li
        class="user-block__item"
      >
        <div
          class="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            width="63"
          />
        </div>
      </li>
      <li
        class="user-block__item"
      >
        <a
          class="user-block__link"
          href="/"
        >
          Sign out
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
